src/System.Private.Xml/src/System/Xml/Xsl/Runtime/XmlCollation.Windows.cs

```diff
@@ -0,0 +1,558 @@
+                object.Equals(_cultInfo, that._cultInfo);
+        }
+
+        public override int GetHashCode()
+        {
+            int hashCode = _options;
+            if (_cultInfo != null)
+            {
+                hashCode ^= _cultInfo.GetHashCode();
+            }
+            return hashCode;
+        }
+
+
+        //-----------------------------------------------
+        // Serialization Support
+        //-----------------------------------------------
+
+        // Denotes the current thread locale
+        private const string LOCALE_CURRENT = "<!-- LOCALE CURRENT -->";
+
+        internal void GetObjectData(BinaryWriter writer)
+        {
+            // NOTE: For CultureInfo we serialize only LCID. It seems to suffice for our purposes.
+            Debug.Assert(_cultInfo == null || _cultInfo.Equals(new CultureInfo(_cultInfo.Name)),
+                "Cannot serialize CultureInfo correctly");
+            writer.Write(_cultInfo != null ? _cultInfo.Name : LOCALE_CURRENT);
+            writer.Write(_options);
+        }
+
+        internal XmlCollation(BinaryReader reader)
+        {
+            string cultName = reader.ReadString();
+            _cultInfo = (cultName != LOCALE_CURRENT) ? new CultureInfo(cultName) : null;
+            _options = new Options(reader.ReadInt32());
+            _compops = _options.CompareOptions;
+        }
+
+        //-----------------------------------------------
+        // Compare Properties
+        //-----------------------------------------------
+
+        internal bool UpperFirst
+        {
+            get { return _options.UpperFirst; }
+        }
+
+        internal bool EmptyGreatest
+        {
+            get { return _options.EmptyGreatest; }
+        }
+
+        internal bool DescendingOrder
+        {
+            get { return _options.DescendingOrder; }
+        }
+
+        internal CultureInfo Culture
+        {
+            get
+            {
+                // Use default thread culture if this.cultinfo = null
+                if (_cultInfo == null)
+                    return CultureInfo.CurrentCulture;
+
+                return _cultInfo;
+            }
+        }
+
+        //-----------------------------------------------
+        //
+        //-----------------------------------------------
+
+        /// <summary>
+        /// Create a sort key that can be compared quickly with other keys.
+        /// </summary>
+        internal XmlSortKey CreateSortKey(string s)
+        {
+            SortKey sortKey;
+            byte[] bytesKey;
+            int idx;
+
+            sortKey = Culture.CompareInfo.GetSortKey(s, _compops);
+
+            // Create an XmlStringSortKey using the SortKey if possible
+#if DEBUG
+            // In debug-only code, test other code path more frequently
+            if (!UpperFirst && DescendingOrder)
+                return new XmlStringSortKey(sortKey, DescendingOrder);
+#else
+            if (!UpperFirst)
```

Comment:
I couldn't find a way to set UpperFirst to true. As far as I could understand from parsing the code, this is always using `CompareOptions.Ordinal` which will lead to UpperFirst being false here. I'm not 100% on this though.